src/storage/DataAccessorBasedStore.ts

```diff
@@ -347,8 +348,22 @@ export class DataAccessorBasedStore implements ResourceStore {
    * @param slug - Slug to use for the new URI.
    */
   protected createURI(container: ResourceIdentifier, isContainer: boolean, slug?: string): ResourceIdentifier {
-    return { path:
-        `${ensureTrailingSlash(container.path)}${slug ? trimTrailingSlashes(slug) : uuid()}${isContainer ? '/' : ''}` };
+    let path = ensureTrailingSlash(container.path);
+    path += slug ? this.cleanSlug(slug) : uuid();
+    path += isContainer ? '/' : '';
+    return { path };
```

Comment:
Let's have 3 different parts and concatenate them in the end.

```suggestion
    const base = ensureTrailingSlash(container.path);
    const name = slug ? this.cleanSlug(slug) : uuid();
    const suffix += isContainer ? '/' : '';
    return { path: `${base}${name}${suffix}` };
```

Reply:
And actually, there's another case we should test, namely where `slug === '////'` or any arbitrary number of slashes, which would result in an empty slug after cleaning. So what you want is something like

```javascript
slug && this.cleanSlug(slug) || uuid()
```